Sources/TSCUtility/Netrc.swift

```diff
@@ -0,0 +1,166 @@
+import Foundation
+import TSCBasic
+
+
+/// Supplies `Authorization` header, typically to be appended to `URLRequest`
+public protocol AuthorizationProviding {
+    /// Optional `Authorization` header, likely added to `URLRequest`
+    func authorization(for url: Foundation.URL) -> String?
+}
+
+extension AuthorizationProviding {
+    public func authorization(for url: Foundation.URL) -> String? {
+        return nil
+    }
+}
+
+@available (OSX 10.13, *)
+/// Container of parsed netrc connection settings
+public struct Netrc: AuthorizationProviding {
+    
+    /// Representation of `machine` connection settings & `default` connection settings.  If `default` connection settings present, they will be last element.
+    public let machines: [Machine]
+    
+    private init(machines: [Machine]) {
+        self.machines = machines
+    }
+    
+//    /// Testing API.  Not for productive use.
+//    /// See:  [Remove @testable from codebase](https://github.com/apple/swift-package-manager/commit/b6349d516d2f9b2f26ddae9de2c594ede24af7d6)
+//    public static var _mock: Netrc? = nil
+        
+    /// Basic authorization header string
+    /// - Parameter url: URI of network resource to be accessed
+    /// - Returns: (optional) Basic Authorization header string to be added to the request
+    public func authorization(for url: Foundation.URL) -> String? {
+        guard let index = machines.firstIndex(where: { $0.name == url.host }) ?? machines.firstIndex(where: { $0.isDefault }) else { return nil }
+        let machine = machines[index]
+        let authString = "\(machine.login):\(machine.password)"
+        guard let authData = authString.data(using: .utf8) else { return nil }
+        return "Basic \(authData.base64EncodedString())"
+    }
+    
+    ///
+    /// - Parameter fileURL: Location of netrc file, defaults to `~/.netrc`
+    /// - Returns: `Netrc` container with parsed connection settings, or error
+    public static func load(fromFileAtPath filePath: AbsolutePath? = nil) -> Result<Netrc, Netrc.Error> {
+                
+        let filePath = filePath ?? AbsolutePath("\(NSHomeDirectory())/.netrc")
```

Comment:
I know this probably wasn't test on Windows, but just in case, how is this supposed to be handled there? Should this have something like `#if !os(Windows)` or maybe a `FIXME` comment to help the efforts in the ongoing Windows port?

Reply:
probably both.  I don't have a test environment set up for win now, but the resolution would be the win-specific user directory.

Reply:
Would you be able to resolve formatting issues with newlines at the start of declarations to make it consistent with the rest of the codebase?